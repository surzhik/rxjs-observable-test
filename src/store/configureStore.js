import { createStore, applyMiddleware } from 'redux';
import { createEpicMiddleware } from 'redux-observable';
import thunk from 'redux-thunk';
import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';
import { name, version } from '../../package.json';
import { rootReducer } from '../reducers';
import { rootEpic } from '../epics';
import createHelpers from './createHelpers';
import createLogger from './logger';

export default function configureStore(initialState, helpersConfig) {
  const helpers = createHelpers(helpersConfig);
  const middleware = [thunk.withExtraArgument(helpers)];
  const epicMiddleware = createEpicMiddleware();

  let enhancer;
  middleware.push(epicMiddleware);
  if (__DEV__) {
    middleware.push(createLogger());

    // https://github.com/zalmoxisus/redux-devtools-extension#14-using-in-production
    const composeEnhancers = composeWithDevTools({
      // Options: https://github.com/zalmoxisus/redux-devtools-extension/blob/master/docs/API/Arguments.md#options
      name: `${name}@${version}`,
    });

    // https://redux.js.org/docs/api/applyMiddleware.html
    enhancer = composeEnhancers(applyMiddleware(...middleware));
  } else {
    enhancer = applyMiddleware(...middleware);
  }

  // https://redux.js.org/docs/api/createStore.html
  const store = createStore(rootReducer, initialState, enhancer);
  epicMiddleware.run(rootEpic);
  // Hot reload reducers (requires Webpack or Browserify HMR to be enabled)
  if (__DEV__ && module.hot) {
    module.hot.accept('../reducers', () =>
      // eslint-disable-next-line global-require
      store.replaceReducer(require('../reducers').default),
    );
  }

  return store;
}
