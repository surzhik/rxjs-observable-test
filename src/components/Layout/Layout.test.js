/* eslint-env jest */
/* eslint-disable padded-blocks, no-unused-expressions */

import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import App from '../App';
import Layout from './Layout';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);
const initialState = {
  display: {
    isPinging: false,
    init: false,
    monitor: {
      temperature: null,
      airPressure: null,
      humidity: null,
    },
    history: {
      temperature: null,
      airPressure: null,
      humidity: null,
    },
  },
  modeSwitch: {
    nightMode: false,
  },
};

describe('Layout', () => {
  test('renders children correctly', () => {
    const store = mockStore(initialState);
    const wrapper = renderer
      .create(
        <App
          context={{
            insertCss: () => {},
            fetch: () => {},
            pathname: '',
            store,
          }}
        >
          <Layout>
            <div />
            <div />
            <div />
          </Layout>
        </App>,
      )
      .toJSON();

    expect(wrapper).toMatchSnapshot();
  });
});
