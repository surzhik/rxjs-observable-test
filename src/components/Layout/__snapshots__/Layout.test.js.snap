// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders children correctly 1`] = `
<div>
  <div
    className="headerHolder"
  >
    <div
      className="switchHolder"
    >
      <button
        aria-checked={false}
        checked={false}
        className="rc-switch"
        defaultChecked={false}
        onChange={[Function]}
        onClick={[Function]}
        onKeyDown={[Function]}
        onMouseUp={[Function]}
        role="switch"
        type="button"
      >
        <span
          className="rc-switch-inner"
        />
      </button>
    </div>
  </div>
  <div />
  <div />
  <div />
  <div
    className="footerHolder"
  >
    <div
      className="container"
    >
      Â© 2019
      <a
        className="link"
        href="https://github.com/surzhik/rxjs-observable-test"
        onClick={[Function]}
      >
        Ivan Turuk
      </a>
    </div>
  </div>
</div>
`;
